app/plugin.go

```diff
@@ -270,6 +268,8 @@ func (a *App) ShutDownPlugins() {
 
 	a.RemoveConfigListener(a.Srv.PluginConfigListenerId)
 	a.Srv.PluginConfigListenerId = ""
+	a.Srv.PluginsLock.Lock()
+	defer a.Srv.PluginsLock.Unlock()
 	a.Srv.PluginsEnvironment = nil
```

Comment:
2/5 that we should use use `SetPluginsEnvironment(nil)` here to avoid more manuell locking.